app/serializers/dist/json_serializer.rb

```diff
@@ -70,5 +72,61 @@ def serialize_nested(connection)
         name: connection.name,
       }
     end
+
+    def build_mapping_blocks(mapping_rules)
+      all_mapping_blocks = []
+      puts "Generating mappings ..."
+      mapping_count = 0
+      Integration.all.each do |integration|
+        [true, false].each do |from_shopify|
+          mapping_rule_block = mapping_rules.where(
+            integration_id: integration.id,
+            from_shopify: from_shopify,
+          )
+          next if mapping_rule_block.count.zero?
+
+          mappings = []
+          Category.verticals.each do |vertical|
+            mappings << MappingBuilder.build_one_to_one_mappings_for_vertical(
+              mapping_rules: mapping_rule_block,
+              vertical: vertical,
+            )
+          end
+          processed_mappings = mappings.flatten.compact
```

Comment:
AFAICT `compact` isn't needed here

```suggestion
          mappings = Category.verticals.flat_map do |vertical|
            MappingBuilder.build_one_to_one_mappings_for_vertical(mapping_rules:, vertical:)
          end
```